ui/src/main/java/edu/wpi/grip/ui/codeGeneration/Exporter.java

```diff
@@ -0,0 +1,143 @@
+
+  public String stepNames(Pipeline pipeline) {
+    StringBuilder out = new StringBuilder();
+    for (Step step : getSteps(pipeline)) {
+      out.append(step.getOperationDescription().name() + " \n");
+      out.append(getInputNames(step));
+      out.append(getOutputNames(step));
+    }
+    return out.toString();
+  }
+
+  public List<Step> getSteps(Pipeline pipeline) {
+    return pipeline.getSteps();
+  }
+
+  public String getInputNames(Step step) {
+    StringBuilder out = new StringBuilder();
+    for (InputSocket input : step.getInputSockets()) {
+      String type = TemplateMethods.parseSocketType(input);
+      String name = TemplateMethods.parseSocketName(input);
+      String value = TemplateMethods.parseSocketValue(input);
+      out.append(createClass(type, name, value) + " \n");
+      out.append(type + "\n");
+    }
+    out.append(step.getOperationDescription().summary() + "\n");
+    return out.toString();
+  }
+
+  public static String createClass(String type, String name, String value) {
+    if (type.contains("Enum")) {
+      return "int " + name + " = Imgproc." + value;
+    } else if (type.equals("Type")) {
+      return "String " + name + " = \"" + value + "\"";
+    } else if (type.equals("List")) {
+      StringBuilder out = new StringBuilder();
+      out.append("double min" + name + " = " + value.substring(1, value.indexOf(",")) + "\n");
+      out.append("double max" + name + " = " + value.substring(value.indexOf(",") + 1, value.lastIndexOf
+          ("]")));
+      return out.toString();
+    } else {
+      return type + " " + name + " = " + value;
+    }
+
+  }
+
+
+  public String getOutputNames(Step step) {
+    StringBuilder out = new StringBuilder();
+    for (OutputSocket output : step.getOutputSockets()) {
+      out.append("    " + output.getSocketHint().getType().getSimpleName() + ": " + output.getSocketHint()
+          .getIdentifier() +
+          " \n");
+    }
+    return out.toString();
+  }
+
+  public void export(Pipeline pipeline, Language lang, File dir, boolean loadLib) {
+    TPipeline tPipeline = new TPipeline(pipeline);
+    TemplateMethods tempMeth = new TemplateMethods();
+    VelocityContext context = new VelocityContext();
+    context.put("pipeline", tPipeline);
+    context.put("tMeth", tempMeth);
+    context.put("fileName", dir.getName().substring(0, dir.getName().lastIndexOf(".")));
+    context.put("loadLib", loadLib);
+    StringBuilder templateDirBuilder = new StringBuilder();
+    templateDirBuilder.append("src/main/resources/edu/wpi/grip/ui/templates/");
+    switch (lang) {
+      case JAVA:
+        templateDirBuilder.append("java");
+        break;
+      case PYTHON:
+        templateDirBuilder.append("python");
+        break;
+      case CPP:
+        templateDirBuilder.append("cpp");
+        break;
+      default:
+        throw new IllegalArgumentException(lang.toString() + " is not a supported language for code generation.");
+    }
```

Comment:
Perhaps it would make sense to move some of the template directory location logic inside of the Language enum? That would allow us to avoid the switch statement.